picasso/src/main/java/com/squareup/picasso/Cache.java

```diff
@@ -15,56 +15,105 @@
  */
 package com.squareup.picasso;
 
+import android.content.Context;
 import android.graphics.Bitmap;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.LruCache;
 
-/**
- * A memory cache for storing the most recently used images.
- * <p>
- * <em>Note:</em> The {@link Cache} is accessed by multiple threads. You must ensure
- * your {@link Cache} implementation is thread safe when {@link Cache#get(String)} or {@link
- * Cache#set(String, android.graphics.Bitmap)} is called.
- */
-interface Cache {
-  /** Retrieve an image for the specified {@code key} or {@code null}. */
-  Bitmap get(String key);
-
-  /** Store an image in the cache for the specified {@code key}. */
-  void set(String key, Bitmap bitmap);
+import static com.squareup.picasso.Utils.KEY_SEPARATOR;
 
-  /** Returns the current size of the cache in bytes. */
-  int size();
+/** A memory cache which uses a least-recently used eviction policy. */
+final class Cache {
```

Comment:
this diff is a mess, but this is a pure copy-paste from LruCache.

Reply:
I think we should call this `PlatformLruCache` in case we decide to re-add the `Cache` interface in the future it will prevent another rename. I'm not 100% convinced that pluggable caches are a bad thing, I just think we might as well hide it for now until a clear use case arises.

Reply:
sounds good.